### PR TITLE
Create Report.md

### PR DESCRIPTION
**Pull request by ghost**
need to add outputs and explanations, but the logic for what will/won't be detected is explained.

### CODE REVIEW

File: Report.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Cybersecurity Homework 3
+## Port Scanner Detector
+### Ryan Vacca, Matthew Moltzau, Julia Vrooman
+
+## Lan setup
+
+Lan is set up using a virtual ethernet (vnet0) within the host machine. There are 2 Kali Linux machines running on VMWare on the host machine. Each Kali Linux instance is set to connect to the network through the virtual network established by the host. The network IP is 192.168.10.0\24. Therefore, the Kali Linux machines will be assigned IP addresses between 192.168.10.1 and 192.168.10.254, as .0 is reserved for the network and .255 is reserved for broadcast.
```
Comment (moltzaum) [inline Report.md:7]: 192.168.10.0\24 should be 192.168.10.0/24

File: Report.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Code description
+### PortScanner.py
+
+PortScanner.py reads in the target machine's IP address, followed by an integer value which represents the wait time between scans. Then, a TCP socket connection is created between the machine running PortScanner.py and the target machine's IP address and socket 0 to begin with. If the current port on the target's machine is open for TCP connections, the PortScanner's tcpScanner function returns true and then closes the established TCP socket connection. If no connection is established, it returns false. Upon return of a boolean, either the found responsive port is printed to the host machine's console or nothing is, and then the port number is increased after the wait time has elapsed to try the next port number on the target machine. Every time an open port is found, that port's number is printed to the screen of the host machine. This continues to iterate through all port numbers from 0 to 1023 until the POrtScanner.py program is exited with Ctrl^C.
```
Comment (moltzaum) [inline Report.md:12]: I think "represents the wait time between scans" should be "represents the wait time between scans in milliseconds".

With the current code, `PortScanner.py`actually doesn't exit with Ctrl^C. The filename also has a typo.

File: Report.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+### Detector (detector.py, main.py, psocket.py, sniffer.py)
+
+The scanner detector is initiated by running main.py with python3. This file creates a queues in which to hold incoming connection requests and the channels of these requests, as well as a table in which to hold this data. The main.py file initiates the detector and creates separate threads to sniff the data, dissect what comes in, and perform calculations to detect a port scanner. As data comes in and is added to the table, the main.py function displays to the console what the change in the table's size has been since its last addition and updates the user as to the total number of entries in the table.
```
Comment (moltzaum) [inline Report.md:16]: "a queues" should be "queues"

"initiated" and "initiates" are used within the span of a few sentences, which feels repetitive.

File: Report.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+The sniffer.py file sniffs the data coming in and places it on the data queue to be processed by the dissector. It also passes the channel through which this data was received through to the detector. This is necessary so that incoming packets do not get lost while the sniffer attempts to process and decide if past packets need to be added to the table. The queue allows the data to be processed without causing a bottleneck or collisions as simply adding data to the queue does not block the data stream of the thread. This also prevents waiting for database access, as the queue will sort data into an easily manageable FIFO structure.
```
Comment (moltzaum) [inline Report.md:18]: "It also passes the channel through which this data was received through to the detector." There is a separate thread for this, though they are in the same file. Not very clear with this statement.

"This also prevents waiting for database access" This is correct, but I wouldn't call it a database.

**Review by moltzaum — COMMENTED** [review]
Looks good, but I have a few comments. We can pull this in and make the changes, or you can update the document.

File: Report.md (hunk 3 above)
Comment (ghost) [reply Report.md:16]: Matt you can make whatever changes you want idk why you're making petty corrections on a doc that I put up for review.

Comment (ghost) [reply Report.md:16]: If you want to make changes, make them. I'm not going in to make corrections for you when I still have to get a bunch of results.

**Pull request merged**